exercises/practice/dnd-character/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+pub fn modifier(_ability_score: u8) -> f32 {
+    unimplemented!("Implement a utility function that calculates the ability modifier for a given ability score")
+}
+
+pub struct Character {}
+
+impl Character {
+    pub fn new() -> Self {
+        unimplemented!("Create a function that generates a new random character with the following ability scores and hitpoints")
+    }
+
+    pub fn roll_four_dice() -> [u8; 4] {
+        unimplemented!("Create a utility function that rolls four dice")
+    }
+
+    pub fn calculate_ability_score(_ability_dice_rolls: [u8; 4]) -> u8 {
+        unimplemented!("Calculate the ability score from the given rolled dice")
+    }
+
+    pub fn strength(&self) -> u8 {
+        unimplemented!("Return the strength score of the character")
+    }
+
+    pub fn dexterity(&self) -> u8 {
+        unimplemented!("Return the dexterity score of the character")
+    }
+
+    pub fn constitution(&self) -> u8 {
+        unimplemented!("Return the constitution score of the character")
+    }
+
+    pub fn intelligence(&self) -> u8 {
+        unimplemented!("Return the intelligence score of the character")
+    }
+
+    pub fn wisdom(&self) -> u8 {
+        unimplemented!("Return the wisdom score of the character")
+    }
+
+    pub fn charisma(&self) -> u8 {
+        unimplemented!("Return the charisma score of the character")
+    }
+
+    pub fn hitpoints(&self) -> u8 {
+        unimplemented!("Return the hitpoints of the character")
+    }
```

Comment:
I'm slightly torn whether these methods are a good or a bad idea. 

- The plus: users are free to implement the struct as they see fit
- The downside: it seems like quite boilerplatey

Reply:
true about the downside, but I think it is challenging to avoid boilerplate code in this exercise, given its classic object-oriented nature. 

I personally enjoy having these getter methods, as it's allowing me to access attributes like `Character.strength()`. I included these methods in the stub file to ensure that the rust-analyzer doesn't show any errors when processing the test file.

on the other hand it is a perfect spot to write a macro, as I just did in the updated example file.

Reply:
I'll let @petertseng weigh in. The methods seem like overkill to me over regular struct access, but 🤷